test/document.test.js

```diff
@@ -14296,6 +14296,93 @@ describe('document', function() {
 
     delete mongoose.Schema.Types.CustomType;
   });
+
+  it('supports schemaFieldsOnly option for toObject() (gh-15258)', async function() {
+    const schema = new Schema({ key: String }, { discriminatorKey: 'key' });
+    const subschema1 = new Schema({ field1: String });
+    const subschema2 = new Schema({ field2: String });
+
+    const Discriminator = db.model('Test', schema);
+    Discriminator.discriminator('type1', subschema1);
+    Discriminator.discriminator('type2', subschema2);
+
+    const type1Key = 'type1';
+    const type2Key = 'type2';
```

Comment:
For consistency, this could also be used in the `.discriminator` calls

Reply:
I went the other approach in [872a2e7](https://github.com/Automattic/mongoose/pull/15259/commits/872a2e7dc170fbfbbc4d10d52969ee90a8c9c62d) and removed `type1Key` and `type2Key` variables for consistency. I tend to dislike adding variable definitions for strings in tests because they add SLOC without improving readability IMO